### DISCUSSION TITLE
FastAPI accepting empty string value for a mandatory/required field

### DISCUSSION THREAD

**Discussion by nistha314e** [fastapi/fastapi#9327] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

app.post("/test")
def get_data(status: str):
   return status
```

### Description

I have defined a POST request and have passed a string parameter. When I run this API using the Swagger/docs, and do not pass any value to the **status** field, it does not allow me to run the API since the status field is a mandatory field.

But if I call this API using the **curl** request, it allows an empty parameter value to pass to it. It gives a 200 response and empty value is passed to the API even if it contains a required field.

`curl -X 'POST' 'http://localhost:8000/api/v1/test?status=' -H 'accept: application/json'`

I tried using: 
status: str    -> Passed an empty value
status: str = Query(...)    -> Passed an empty value
status: str = Required.      -> Passed an empty value

How can an empty value be passed even if the parameter is a mandatory field? 

### Operating System

Linux, macOS

### FastAPI Version

0.95.0

### Python Version

3.9.16

**Comment by abhint** [comment]
```py 
from fastapi import FastAPI

app = FastAPI()


@app.get("/items/{item_id}")
async def read_user_item(item_id: str, needy: str):
    item = {"item_id": item_id, "needy": needy}
    return item
```

https://fastapi.tiangolo.com/tutorial/query-params/?h=required#required-query-parameters

**Reply by nistha314e** [reply]
That's what I am saying, it is a required field and still accepting empty values. 

curl -X 'POST' 'localhost:8000/api/v1/test?**needy=**' -H 'accept: application/json'
This is accepting empty values.

**Reply by abhint** [reply]
```py 
from typing import Annotated

from fastapi import FastAPI, Query
from pydantic import Required

app = FastAPI()


@app.post("/test/")
def get_data(status: Annotated[str, Query(min_length=3)] = Required):
   return {"status" : status}
```

**Reply by nistha314e** [reply]
Yeah I tried this one, it does work. 
But my concern is that besides being a required field, how can FastAPI accept empty values. It shouldn't right? If it is not accepting an empty value from Swagger UI docs, then why accept it through curl command?

**Reply by abhint** [reply]
I don't know why that happens. 💭 

```py
@app.post("/test/")
async def get_data(status: str):
    if len(status)==0: return {"error": "empty string"}
    return {"status": status}
```

**Comment by ebottos94** [comment, marked as answer]
@Nistha314e in Swagger UI there is an input for your query parameter and when calling 'execute' check if the input is empty, if is required, and if it's empty doesn't execute the function. But if you call the endpoint from an external request with another client it works if you don't set validation length because this 'http://localhost:8000/api/v1/test?status=' is simply a path with status equal to ''.

**Reply by Ark-kun** [reply]
>in Swagger UI there is an input for your query parameter and when calling 'execute' check if the input is empty, if is required, and if it's empty doesn't execute the function

Yes. And it's a bug. Empty string is still a string. The Swagger UI should allow accepting it. Especially for path parameters.
